### PR TITLE
Add AWS support

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/4795

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,4 +3,39 @@
...
+$ helm install external-dns-app -f values.yaml
+```
+
+Deployment to Tenant Clusters is handled by [app-operator](https://github.com/giantswarm/app-operator).
+
+## Configuration
+
+Configuration options are documented in [Configuration.md](helm/external-dns-app/Configuration.md) document.
+
+## Release Process
```
Comment (pipo02mix) [inline README.md]: Is this something we have for all apps? I would rather to have it just one place in common, so easy to maintain.

File: helm/external-dns-app/Configuration.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# external-dns
+
+This chart installs `external-dns` as managed applications. ExternalDNS synchronizes exposed Kubernetes Services and Ingresses with DNS providers.
```
Comment (pipo02mix) [inline helm/external-dns-app/Configuration.md]:
```suggestion
This chart installs `external-dns` as managed application. ExternalDNS synchronizes exposed Kubernetes Services and Ingresses with DNS providers.
```

File: helm/external-dns-app/templates/deployment.yaml [hunk 3]
```diff
@@ -22,6 +22,9 @@ spec:
         giantswarm.io/service-type: "managed"
         app: {{ .Values.name }}
       annotations:
+        {{- if eq .Values.provider "aws" }}
+        iam.amazonaws.com/role: {{ .Values.clusterID}}-Route53Manager-Role
```
Comment (pipo02mix) [inline helm/external-dns-app/templates/deployment.yaml]:
```suggestion
        iam.amazonaws.com/role: {{ .Values.role}}
```
What if we do it more universal so we can reuse for something else?
(adding cluster id with the statis text on value)

File: helm/external-dns-app/values.yaml [hunk 4]
```diff
@@ -8,7 +8,12 @@ namespace: kube-system
 userID: 65534
 groupID: 65534
 
-domainFilter: gigantic.io
+baseDomain: gigantic.io
+# clusterID is dynamic environment value, calculated after cluster creation
+# applies only to Giant Swarm clusters
+clusterID: en2jo
```
Comment (pipo02mix) [inline helm/external-dns-app/values.yaml]:
```suggestion
role: en2jo-Route53Manager-Role
```

File: helm/external-dns-app/Configuration.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+
+## Configuration
+
+The following table lists the configurable parameters of the external-dns chart, its dependencies and default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`baseDomain` | Cluster base domain. `external-dns` applies only to `aws` privder | 'aws'
+`clusterID` | Cluster identifier. Applies only to Giant Swarm managed clusters | 'testid'
```
Comment (pipo02mix) [inline helm/external-dns-app/Configuration.md]:
```suggestion
`role` | IAM Role used to modify route 53 | 'testid-Route53Manager-Role'
```

File: helm/external-dns-app/templates/deployment.yaml (hunk 3 above)
Comment (corest) [reply helm/external-dns-app/templates/deployment.yaml]: We don't have role anywhere. We have `clusterID` from here https://github.com/giantswarm/cluster-operator/pull/731 
We could define value in template, but I don't consider this app to be reusable anywhere else for now

Comment (corest) [reply helm/external-dns-app/templates/deployment.yaml]: Also we're using `clusterID` in `kiam` already.

I'll add an issue to wrap it in a different way as a next step

File: README.md (hunk 1 above)
Comment (rossf7) [reply README.md]: @pipo02mix Yes I've been adding this README format to all the new app repos.

I see your point about making it DRY with a link but the steps are very generic and unlikely to change.

File: helm/external-dns-app/templates/deployment.yaml (hunk 3 above)
Comment (corest) [reply helm/external-dns-app/templates/deployment.yaml]: https://github.com/giantswarm/giantswarm/issues/7397

File: README.md (hunk 1 above)
Comment (pipo02mix) [reply README.md]: ok good to know :)

File: helm/external-dns-app/templates/deployment.yaml (hunk 3 above)
Comment (pipo02mix) [reply helm/external-dns-app/templates/deployment.yaml]: ok

**Pull request merged**